### PR TITLE
Add `where=` arg to jax.nn.{softmax, log_softmax, normalize}.

### PR DESCRIPTION
**Pull request by rjagerman**
This change adds a `where=` argument (analogous to `jnp.sum`, `jnp.mean`, etc) to `jax.nn.{softmax, log_softmax, normalize}` that can be used to specify which elements to include in the calculation. This is useful for situations with padding where certain elements must be excluded when calling the `softmax`, `log_softmax` or `normalize` functions.

### CODE REVIEW

File: jax/_src/nn/functions.py [hunk 1]
```diff
@@ -275,11 +276,16 @@ def log_softmax(x: Array, axis: Optional[Union[int, Tuple[int, ...]]] = -1) -> A
     x : input array
     axis: the axis or axes along which the :code:`log_softmax` should be
       computed. Either an integer or a tuple of integers.
+    where: Elements to include in the log_softmax.
```
Comment (jakevdp) [inline jax/_src/nn/functions.py:279]: `jax.numpy.max` requires `initial` to be explicitly passed when `where` is used – I think it would make sense to do the same for `log_softmax` and `softmax`. For `normalize`, it's probably not necessary because it has a natural identity.

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/nn/functions.py (hunk 1 above)
Comment (rjagerman) [reply jax/_src/nn/functions.py]: Sounds good. I added `initial` as an argument for `softmax` and `log_softmax`.

**Review by jakevdp — APPROVED** [review]
Looks good!

File: tests/nn_test.py [hunk 2]
```diff
@@ -123,6 +123,28 @@ def testHardTanhMemory(self):
...
+
+    self.assertAllClose(out_masked, out_filtered)
+
+  def testNormalizeWhereMask(self):
+    x = jnp.array([5.5, 1.3, -4.2, 0.9])
+    m = jnp.array([True, False, True, True])
+    x_filtered = jnp.take(x, jnp.array([0, 2, 3]))
+
+    out_masked = jnp.take(nn.normalize(x, where=m), jnp.array([0, 2, 3]))
+    out_filtered = fn(x_filtered)
```
Comment (jakevdp) [inline tests/nn_test.py:144]: Looks like `fn` is undefined

Comment (rjagerman) [reply tests/nn_test.py]: Ah, missed that one. Should be fixed now.
